File: bitcode_android/MenuDemo2/app/src/main/java/in/kaligotla/menudemo2/view/MainActivity.java
package in.kaligotla.menudemo2.view;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import in.kaligotla.menudemo2.model.Note;
import in.kaligotla.menudemo2.viewModel.NoteViewModel;
import in.kaligotla.menudemo2.R;
import in.kaligotla.menudemo2.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    ActivityMainBinding binding;
    NoteViewModel noteViewModel;
    NotesFragment notesFragment;
    boolean showingArchived = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        noteViewModel = new ViewModelProvider(this).get(NoteViewModel.class);
        notesFragment = (NotesFragment) getSupportFragmentManager().findFragmentById(R.id.notesFragment);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.options_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.action_add) {
            showAddNoteDialog();
            return true;
        } else if (id == R.id.action_toggle_archived) {
            showingArchived = !showingArchived;
            item.setTitle(showingArchived ? "Hide Archived" : "Show Archived");
            if (notesFragment != null) notesFragment.toggleArchived();
            return true;
        } else if (id == R.id.action_sync) {
            syncNotesWithCloud();
            return true;
        } else if (id == R.id.action_export) {
            exportNotes();
            return true;
        } else if (id == R.id.action_settings) {
            openSettings();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private void showAddNoteDialog() {
        EditText input = new EditText(this);
        new AlertDialog.Builder(this)
                .setTitle("New Note")
                .setView(input)
                .setPositiveButton("Add", (dialog, which) -> {
                    String text = input.getText().toString();
                    if (!text.isEmpty()) {
                        noteViewModel.insert(new Note(text));
                    }
                })
                .setNegativeButton("Cancel", null)
                .show();
    }

    private void exportNotes() {
        Toast.makeText(this, "Exporting notes...", Toast.LENGTH_SHORT).show();
    }

    private void openSettings() {
        Intent intent = new Intent(android.provider.Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
        intent.setData(Uri.parse("package:" + getPackageName()));
        startActivity(intent);
    }

    private void syncNotesWithCloud() {
        Toast.makeText(this, "Notes synced with cloud", Toast.LENGTH_SHORT).show();
    }

    @SuppressLint("RestrictedApi")
    public static void showIcons(Menu menu) {
        try {
            @SuppressLint("PrivateApi")
            Class<?> menuBuilderClass = Class.forName("com.android.internal.view.menu.MenuBuilder");
            if (menuBuilderClass.isInstance(menu)) {
                @SuppressLint("DiscouragedPrivateApi")
                java.lang.reflect.Method method = menuBuilderClass.getDeclaredMethod("setOptionalIconsVisible", boolean.class);
                method.setAccessible(true);
                method.invoke(menu, true);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: bitcode_android/MenuDemo/app/src/main/java/in/kaligotla/menudemo/MainActivity.kt
package `in`.kaligotla.menudemo

import android.content.ComponentName
import android.content.Intent
import android.os.Bundle
import android.view.ContextMenu
import android.view.Menu
import android.view.MenuItem
import android.view.SubMenu
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import `in`.kaligotla.menudemo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var activityMainBinding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityMainBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(activityMainBinding.root)
        registerForContextMenu(activityMainBinding.textView)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menu.add(0, 1, 0, "Open Detail")
        menu.add(0, 2, 1, "Share Message")
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            1 -> {
                val intent = Intent(this, DetailActivity::class.java)
                intent.putExtra("message", "Hello from MainActivity!")
                startActivity(intent)
                true
            }
            2 -> {
                shareText(activityMainBinding.textView.text.toString())
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    // === Context Menu ===
    override fun onCreateContextMenu(menu: ContextMenu, v: View, menuInfo: ContextMenu.ContextMenuInfo?) {
        menu.setHeaderTitle("Context Menu")
        menu.add(0, 10, 0, "Open Detail")
        menu.add(0, 11, 1, "Share Message")
    }

    override fun onContextItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            10 -> {
                val intent = Intent(this, DetailActivity::class.java)
                intent.putExtra("message", "Hello from MainActivity!" )
                startActivity(intent)
                true
            }
            11 -> {
                shareText(activityMainBinding.textView.text.toString())
                true
            }
            else -> super.onContextItemSelected(item)
        }
    }

    private fun shareText(text: String) {
        val intent = Intent(Intent.ACTION_SEND).apply {
            type = "text/plain"
            putExtra(Intent.EXTRA_TEXT, text)
        }
        val chooser = Intent.createChooser(intent, "Share via")
        startActivity(chooser)
    }
}